### PR TITLE
Ruby Bubble Sort, Project # 1, Ruby - Request for Code review

### PR DESCRIPTION
**Pull request by wickathou**
Features:

A sorting algorithm built with Ruby

### CODE REVIEW

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    (0...length).each do |n|
+      key = arr[n]
+      if arr[n + 1] < arr[n]
+        arr[n] = arr[n + 1]
+        arr[n + 1] = key
+      end
+    end
+    ite += 1
+  end
+  puts arr
```
Comment (Ispirett) [inline bubble_sort.rb]: Please remove puts and just return the array

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  length = arr.size - 1
+  while ite < length
+    (0...length).each do |n|
+      if yield(arr[n], arr[n + 1]).positive?
+        arr[n], arr[n + 1] = arr[n + 1], arr[n]
+      end
+    end
+    ite += 1
+  end
+  puts arr
```
Comment (Ispirett) [inline bubble_sort.rb]: Please remove puts and just return the array.

**Review by Ispirett — CHANGES_REQUESTED** [review]
## Status: Changes Required
Excellent Work!! @wickathou  @Grifo89  with this project so far :+1:  :100: 

image

Currently, your project returns nil 
 - [x] Checkout comments below to solve this issue.

 - [x] Please add stickler to your project, follow the instructions from pathwright.

**please rememeber to us more descriptive commit messages.**

File: bubble_sort.rb (hunk 1 above)
Comment (wickathou) [reply bubble_sort.rb]: Done

File: bubble_sort.rb (hunk 2 above)
Comment (wickathou) [reply bubble_sort.rb]: Done

File: bubble_sort.rb (hunk 1 above)
Comment (wickathou) [reply bubble_sort.rb]: Done

**Comment by wickathou** [comment]
We had to use Rubocop as we were having issues with Stickler.

File: bubble_sort.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  ite = 0
+  length = arr.size - 1
+  while ite < length
+    (0...length).each do |n|
+      key = arr[n]
+      if arr[n + 1] < arr[n]
+        arr[n] = arr[n + 1]
+        arr[n + 1] = key
+      end
+    end
```
Comment (wathigo) [inline bubble_sort.rb:13]: **[REQUIRED]** 
We are missing the `end` statement for closing the `while` block.

File: bubble_sort.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+def bubble_sort_by(arr)
+  ite = 0
+  length = arr.size - 1
+  while ite < length
+    (0...length).each do |n|
+      if yield(arr[n], arr[n + 1]).positive?
+        arr[n], arr[n + 1] = arr[n + 1], arr[n]
+      end
+    end
```
Comment (wathigo) [inline bubble_sort.rb:26]: **[REQUIRED]** 
We are missing the `end` statement for closing the `while` block.

File: stickler.yml [hunk 5]
```diff
@@ -0,0 +1,1270 @@
+
```
Comment (wathigo) [inline stickler.yml]: **[REQUIRED]**
Let's the right configurations to the `.stickler.yml` file. Lets remove the html mark up from this file.  
Check out the correct linter configurations for Ruby projects [here](https://github.com/microverseinc/linters-config/tree/master/ruby).

File: rubocop.yml [hunk 6]
```diff
@@ -0,0 +1,1385 @@
+
```
Comment (wathigo) [inline rubocop.yml]: **[REQUIRED]**
Let's include the right linter configurations in `.rubocop.yml` file.
These are the right configs.
```
AllCops:
  Exclude:
    - "bin/*"
    - "db/**/*"
    - "config/**/*"
    - "Guardfile"
    - "Rakefile"
  DisplayCopNames: true

Metrics/LineLength:
  Max: 120
Metrics/MethodLength:
  Include: ["app/controllers/*"]
  Max: 20
Metrics/AbcSize:
  Include: ["app/controllers/*"]
  Max: 30
Metrics/ClassLength:
  Max: 150
Metrics/BlockLength:
  ExcludedMethods: ['describe']

Style/Documentation:
  Enabled: false
Style/ClassAndModuleChildren:
  Enabled: false

Layout/AlignHash:
  EnforcedColonStyle: key
Layout/ExtraSpacing:
  AllowForAlignment: false
Layout/MultilineMethodCallIndentation:
  Enabled: true
  EnforcedStyle: indented
```
Checkout this [repo](https://github.com/microverseinc/linters-config/blob/master/ruby/.rubocop.yml) for the right configurations for this file.

**Review by wathigo — CHANGES_REQUESTED** [review]
### Hi @wickathou  and @Grifo89  :wave:  
Great work implementing the requested changes but we need to fix a few issues before we move on.* 

* See the issues pointed out in the comments below :point_down:  

**After implementing the changes, submit another request for review.**

_Have a great weekend :smile:  
Reviewed by: **Simon Wathigo**_

File: rubocop.yml (hunk 6 above)
Comment (wickathou) [reply rubocop.yml]: We used directly the ones in this repo
https://github.com/microverseinc/linters-config/tree/master/ruby

File: stickler.yml (hunk 5 above)
Comment (wickathou) [reply stickler.yml]: That's the one we used and it's not working for us. Rubocop is allowed as it's described on the pathwright.

Comment (wathigo) [reply stickler.yml]: Let's remember to name the file correctly. It should be `.stickler.yml` **not** `stickler.yml`

**Comment by Grifo89** [comment]
Hey there, changes and corrections are made.

**Comment by Grifo89** [comment]
[](url)

File: bubble_sort.rb [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+  while ite < length
+    (0...length).each do |n|
+      key = arr[n]
+      if arr[n + 1] < arr[n]
+        arr[n] = arr[n + 1]
+        arr[n + 1] = key
+      end
+    end
+    arr
+  end
```
Comment (samgaco) [inline bubble_sort.rb:15]: :+1:

**Review by samgaco — APPROVED** [review]
Everything is fine, you may move on :+1: 

------------------------------------------

* Friendly reminder: You should use [Github flow](https://microverse.pathwright.com/library/microverse-onboarding/86777/path/step/53987922/) in this project. However, changing it now would take too much time. Please make sure you use [Github Flow](https://microverse.pathwright.com/library/microverse-onboarding/86777/path/step/53987922/) in the next project. 
------------------------------------------

* ###### STATUS OF THE PROJECT: APPROVED :heavy_check_mark:

**Pull request merged**